### PR TITLE
operators, quoted strings, etc

### PR DESCRIPTION
**Pull request by wxwisiasdf**

### CODE REVIEW

**Review by adamtherookie — APPROVED** [review]

**Review by NineSilver — CHANGES_REQUESTED** [review]
Using abort() isn't a good idea if we want to report errors to the user.
Because we are using string views it isn't necessary to check for '\0'.
I don't see the point on treating comments as tokens, as we aren't parsing nor making anything with them.

**Comment by wxwisiasdf** [comment]
> Using abort() isn't a good idea if we want to report errors to the user. Because we are using string views it isn't necessary to check for '\0'. I don't see the point on treating comments as tokens, as we aren't parsing nor making anything with them.

We can save comments for debug information - abort() was just a placeholder until someone made a proper error system - uhm, okey sure(?)

File: src/lexer.c [hunk 1]
```diff
@@ -18,17 +18,29 @@ static char *toktype_str[] = {
...
     [TOK_DOT] = ".",
     [TOK_STAR] = "*",
 
+    [TOK_INC] = "++",
+    [TOK_DEC] = "--",
+
+    [TOK_LOGICAL_AND] = "&",
+    [TOK_LOGICAL_OR] = "|",
+    [TOK_BIT_AND] = "&&",
+    [TOK_BIT_OR] = "||",
```
Comment (jmerono-saez) [inline src/lexer.c:38]: p sure you got these names wrong

File: src/lexer.c [hunk 2]
```diff
@@ -18,17 +18,29 @@ static char *toktype_str[] = {
...
+
     [TOK_INTEGER] = "integer",
     [TOK_STRING] = "string",
     [TOK_CHARACTER] = "character",
-
+    [TOK_COMMENT] = "comment",
```
Comment (jmerono-saez) [inline src/lexer.c:43]: please not

File: src/lexer.c [hunk 3]
```diff
@@ -44,14 +56,119 @@ static str_t read_char(lexer_state_t *state) {
...
+         * strlen(alphanum_set) */
+        /* check that the length is not bigger than the base */
+        int num = (int)((ptrdiff_t)set - (ptrdiff_t)alphanum_set);
+        if (num > base) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
```
Comment (jmerono-saez) [inline src/lexer.c:78]: could be smaller but ig its fine

File: src/lexer.c [hunk 4]
```diff
@@ -60,6 +177,44 @@ static str_t read_identifier(lexer_state_t *state) {
...
+    if (cur(state) == '\0') {
+        if (multiline) {
+            /* throw error: expected termination FOR MULTILINE comment */
+        }
+        end = state->pos;
+    }
+
+    return slice_str(state->src, start, end);
+}
+
```
Comment (jmerono-saez) [inline src/lexer.c:217]: COMMENTS ARE NOT TOKENS

Comment (jmerono-saez) [reply src/lexer.c:217]: in fact, the function reading chars should skip comments entirely

**Review by jmerono-saez — CHANGES_REQUESTED** [review]
holy fuck are you ok

**Review by unsafecast — COMMENTED** [review]
@segfaultdev is pretty much on point.

**Pull request closed without merging**